src/zx/FunctionalityConstruction.cpp

```diff
@@ -223,6 +226,65 @@ FunctionalityConstruction::parseOp(ZXDiagram& diag, op_it it, op_it end,
       addZSpider(diag, target2, qubits, PiExpression(), EdgeType::Hadamard);
       break;
     }
+    case qc::OpType::RZZ: {
+      const auto ctrl = target;
+      const auto target2 = static_cast<zx::Qubit>(p.at(op->getTargets()[1]));
+      const auto param = parseParam(op.get(), 0);
+
+      addZSpider(diag, ctrl, qubits);
+      addZSpider(diag, target2, qubits);
+
+      auto mid_x = diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)),
+                                  zx::VertexType::X);
+      auto mid_z = diag.addVertex(-1, -1, param, zx::VertexType::Z);
+      diag.addEdge(qubits[static_cast<std::size_t>(ctrl)], mid_x);
+      diag.addEdge(qubits[static_cast<std::size_t>(target2)], mid_x);
+      diag.addEdge(mid_x, mid_z);
+      diag.addGlobalPhase(-param / 2.0);
+
+      break;
+    }
+    case qc::OpType::RXX: {
+      const auto ctrl = target;
+      const auto target2 = static_cast<zx::Qubit>(p.at(op->getTargets()[1]));
+      const auto param = parseParam(op.get(), 0);
+
+      addXSpider(diag, ctrl, qubits);
+      addXSpider(diag, target2, qubits);
+
+      auto mid_z = diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)),
+                                  zx::VertexType::Z);
+      auto mid_x = diag.addVertex(-1, -1, param, zx::VertexType::X);
+      diag.addEdge(qubits[static_cast<std::size_t>(ctrl)], mid_z);
+      diag.addEdge(qubits[static_cast<std::size_t>(target2)], mid_z);
+      diag.addEdge(mid_z, mid_x);
+      diag.addGlobalPhase(-param / 2.0);
+      break;
+    }
+    case qc::OpType::RYY: {
+      const auto ctrl = target;
+      const auto target2 = static_cast<zx::Qubit>(p.at(op->getTargets()[1]));
+      const auto param = parseParam(op.get(), 0);
+
+      addXSpider(diag, ctrl, qubits, PiExpression(PiRational(1, 2)));
+      addXSpider(diag, target2, qubits, PiExpression(PiRational(1, 2)));
+
+      addZSpider(diag, ctrl, qubits);
+      addZSpider(diag, target2, qubits);
+
+      auto mid_x = diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)),
+                                  zx::VertexType::X);
+      auto mid_z = diag.addVertex(-1, -1, param, zx::VertexType::Z);
+      diag.addEdge(qubits[static_cast<std::size_t>(ctrl)], mid_x);
+      diag.addEdge(qubits[static_cast<std::size_t>(target2)], mid_x);
+      diag.addEdge(mid_x, mid_z);
```

Comment:
Now it's getting really picky: if you extract the RZZ code into a method, then you can just reuse that method here and safe some code duplication.

I believe it would generally be nicer to have more methods like the `addCnot` one because it creates a better separation of concerns and smaller methods.

Reply:
I would consider two instances of a couple of lines not a bad case of code duplication. Especially since this code is not expected to change anymore.

Reply:
Extracted everything into methods now.